src/lib/conversion/rotation.cpp

```
@@ -62,290 +60,201 @@ get_rot_quaternion(enum Rotation rot)
+			float tmp = x;
 			x = -z;
 			z = y;
 			y = -tmp;
-			return;
 		}
+		break;
 
 	case ROTATION_ROLL_90_PITCH_90: {
-			tmp = x;
+			float tmp = x;
 			x = y;
 			y = -z;
 			z = -tmp;
-			return;
 		}
+		break;
 
 	case ROTATION_ROLL_180_PITCH_90: {
-			tmp = x;
+			float tmp = x;
 			x = -z;
 			y = -y;
 			z = -tmp;
-			return;
 		}
+		break;
 
 	case ROTATION_ROLL_270_PITCH_90: {
-			tmp = x;
+			float tmp = x;
 			x = -y;
 			y = z;
 			z = -tmp;
-			return;
 		}
+		break;
 
 	case ROTATION_ROLL_90_PITCH_180:
 
 	// FALLTHROUGH
 	case ROTATION_ROLL_270_YAW_180: {
-			tmp = y;
+			float tmp = y;
 			x = -x;
 			y = -z;
 			z = -tmp;
-			return;
 		}
+		break;
 
 	case ROTATION_ROLL_270_PITCH_180: {
-			tmp = y;
+			float tmp = y;
 			x = -x;
 			y = z;
 			z = tmp;
-			return;
 		}
+		break;
 
 	case ROTATION_ROLL_90_PITCH_270: {
-			tmp = x;
+			float tmp = x;
 			x = -y;
 			y = -z;
 			z = tmp;
-			return;
 		}
+		break;
 
 	case ROTATION_ROLL_270_PITCH_270: {
-			tmp = x;
+			float tmp = x;
 			x = y;
 			y = z;
 			z = tmp;
-			return;
 		}
+		break;
 
 	case ROTATION_ROLL_90_PITCH_180_YAW_90: {
-			tmp = x;
+			float tmp = x;
 			x = z;
 			z = -y;
 			y = -tmp;
-			return;
 		}
+		break;
 
-	case ROTATION_ROLL_90_PITCH_68_YAW_293: {
-			const float tmpx = x;
-			const float tmpy = y;
-			const float tmpz = z;
-			x = 0.146371f * tmpx + 0.362280f * tmpy - 0.920505f * tmpz;
-			y = -0.344827f * tmpx - 0.853477f * tmpy - 0.390731f * tmpz;
-			z = -0.927184f * tmpx + 0.374607f * tmpy;
-			return;
-		}
+	default:
 
-	case ROTATION_ROLL_90_PITCH_315: {
-			const float tmpx = x;
-			const float tmpy = y;
-			const float tmpz = z;
-			x = 0.707107f * tmpx - 0.707107f * tmpy;
-			y = -tmpz;
-			z = 0.707107f * tmpx + 0.707107f * tmpy;
-			return;
+		// otherwise use full rotation matrix for valid rotations
+		if (rot < ROTATION_MAX) {
+			const matrix::Vector3f r{get_rot_matrix(rot) *matrix::Vector3f{x, y, z}};
+			x = r(0);
+			y = r(1);
+			z = r(2);
 		}
```

What happens in the case of rot > ROTATION_MAX?

---

Silent failure, the same thing that currently happens. At the calibration level (parameters) we catch (report + reset) invalid rotations, so that only leaves low level errors (command line sensor start).